test-app/src/main/java/org/readium/r2/testapp/reader/VisualReaderFragment.kt

```diff
@@ -51,49 +75,205 @@ abstract class VisualReaderFragment : BaseReaderFragment(), VisualNavigator.List
         return binding.root
     }
 
+    /**
+     * When true, the user won't be able to interact with the navigator.
+     */
+    private var disableTouches by mutableStateOf(false)
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         navigatorFragment = navigator as Fragment
 
-        val viewScope = viewLifecycleOwner.lifecycleScope
+        setupObservers()
+
+        childFragmentManager.addOnBackStackChangedListener {
+            updateSystemUiVisibility()
+        }
+        binding.fragmentReaderContainer.setOnApplyWindowInsetsListener { container, insets ->
+            updateSystemUiPadding(container, insets)
+            insets
+        }
+
+        binding.overlay.setContent {
```

Comment:
I added a Jetpack Compose `overlay` view on top of the reader in `VisualReaderFragment`. This is used to display the TTS controls above the publication, but also to prevent user interaction on the pages while the TTS is running.